apache2/waf_lock/waf_lock_external.h

```diff
@@ -0,0 +1,154 @@
+
+#define WAF_LOCK_ERROR                                          1000
+#define WAF_ERROR_LOCK_LINUX_SEM_CREATE_FAIL                    1100
+#define WAF_ERROR_LOCK_LINUX_SEM_OPEN_FAIL                      1101
+#define WAF_ERROR_LOCK_LINUX_SEM_INIT_FAIL                      1102
+#define WAF_ERROR_LOCK_LINUX_SEM_MODIFY_FAIL                    1103
+#define WAF_ERROR_LOCK_LINUX_SEM_DESTROY_FAIL                   1104
+#define	WAF_ERROR_LOCK_LINUX_SEM_SET_PERMISSION_FAIL            1105
+#define	WAF_ERROR_LOCK_LINUX_SEM_GET_USER_FAIL                  1106
+
+#define WAF_ERROR_LOCK_WIN_NAME_INVALID_STRING                  1200
+#define WAF_ERROR_LOCK_WIN_MUTEX_CREATE_FAIL                    1201
+#define WAF_ERROR_LOCK_WIN_ONLY_ONE_LOCK_EXISTS                 1202
+#define WAF_ERROR_LOCK_WIN_GET_MUTEX_FAIL                       1203
+#define WAF_ERROR_LOCK_WIN_RELEASE_MUTEX_FAIL                   1204
+#define WAF_ERROR_LOCK_WIN_CLOSE_MUTEX_FAIL                     1205
+#define WAF_ERROR_LOCK_WIN_DESTROY_NOT_SUPPORT                  1206
+
+#define WAF_LOCK_ERROR_NAME_NULL                                1001
+#define WAF_LOCK_ERROR_NAME_INVALID_STRING                      1002
+#define WAF_ERROR_LOCK_OP_NEGATIVE_VAL                          1003
+#define WAF_LOCK_ERROR_HANDLE_NULL                              1004
+#define WAF_LOCK_ERROR_UNEXPECTED                               1005
+
+
+/**
+ **========================================================
+ ** Waf Lock Handler Definition
+ **========================================================
+ */
+
+/**
+ ** The structure stores the Read/Write locks information.
+ ** In Linux, a set of semaphores is used to implement the Read/Write locks.
+ ** In Windows, The Windows Mutex is used to implement the Read/Write locks.
+ ** @param sem_id: The identifier of semaphore set in Linux.
+ ** @param read_lock_handle: The handler of the read locks in Windows.
+ ** @param write_lock_handle: The handler of the write lock in Windows.
+ */
+struct waf_lock {
+#ifndef _WIN32
+    int sem_id;
+#else
+    HANDLE read_lock_handle;
+    HANDLE write_lock_handle;
+#endif
+};
+
+
+/**
+ **========================================================
+ ** Waf Lock API
+ **========================================================
+ */
+
+/**
+ ** Create a lock.
+ ** @param lock: lock handler.
+ ** return: WAF_LOCK_SUCCESS if successfully created or WAF_LOCK_ERROR if failed.
+ */
+int Waf_createLock(struct waf_lock *lock, const char* lock_name, int lock_name_length, const char* owner);
+
+/**
+ ** Destroy a lock.
+ ** @param lock: lock handler.
+ ** return: WAF_LOCK_SUCCESS if successfully created or WAF_LOCK_ERROR if failed.
+ */
+int Waf_destroyLock(struct waf_lock *lock);
+
+/**
+ ** Close a lock.
+ ** @param lock: lock handler.
+ ** return: WAF_LOCK_SUCCESS if successfully created or WAF_LOCK_ERROR if failed.
+ */
+int Waf_cLoseLock(struct waf_lock *lock);
```

Comment:
Do we need to capitalize 'L' in 'Waf_cLoseLock'

Reply:
will change